odoo/api.py

```diff
@@ -869,6 +869,16 @@ def update(self, records, field, values):
             field_cache = field_cache.setdefault(records.env.cache_key(field), {})
         field_cache.update(zip(records._ids, values))
 
+    def update_keep(self, records, field, values):
+        """ Set the values of ``field`` for several ``records``, but without
+        overwriting the cache.
```

Comment:
I think the naming is confusing, no update ever takes place: it inserts a value if it doesn't exist, if it does it's essentially a no-op.

the easy choice is `insert_if_not_exists`, `insert_ignore` is another option... not sure if those are the best choices, we can discuss it further

Reply:
I also wonder if it's even necessary to call `setdefault` at all, it might be convenient but one of its key points is that it returns the value of the key, which we completely discard here

Reply:
I use `setdefault` for performance reasons: it looks up the key, and updates the dict if necessary without any extra lookup.